Subbus/Classes/Subbus.PersistentEvent.swift

```diff
@@ -0,0 +1,140 @@
+    ///     - replace: Whether this event should replace a previous subscription with the same `id` and `event`.
+    ///     - callback: The handler block for when the event is `Post`ed.
+    static func addSubscription<I, T: PersistentEvent>(id: I, event: T.Type, replace: Bool, callback: @escaping (T) -> PersistentEvent.HandlerResult)
+    
+    /// Posts a `PersistentEvent` to the subscribed receivers.
+    ///
+    /// Parameters:
+    ///     - event: The event to be posted.
+    static func post<T: PersistentEvent>(event: T)
+}
+
+/// A convenience extension to add shortened functions to the `SubbusPersistentEventProtocol`.
+extension SubbusPersistentEventProtocol {
+    /// Adds a new subscription to Subbus with a `PersistentEvent`.
+    ///
+    /// Parameters:
+    ///     - id: The id of the subscriber to this event.  Used for unregistering.
+    ///     - event: The event type to register the subscriber to.
+    ///     - callback: The handler block for when the event is `Post`ed.
+    public static func addSubscription<I, T: PersistentEvent>(id: I, event: T.Type, callback: @escaping (T) -> PersistentEvent.HandlerResult) {
+        addSubscription(id: id, event: event, replace: false, callback: callback)
+    }
+}
+
+/// Conformance to the `SubbusPersistentEventProtocol`.
+extension Subbus: SubbusPersistentEventProtocol {
+    public static func addSubscription<I, T: PersistentEvent>(id: I, event: T.Type, replace: Bool, callback: @escaping (T) -> PersistentEvent.HandlerResult) {
+        let eventType = String(reflecting: T.self)
+        
+        //Register listener
+        guard let subscription = Subscription(identifier: id, eventType: eventType, handler: callback) else { return }
+        addSubscription(subscription, replace: replace)
+        
+        // Post existing events to this new subscription
+        for index in stride(from: unhandledEvents.count - 1, through: 0, by: -1)  {
+            guard let unhandledEvent = unhandledEvents[index] as? T else { continue }
+            
+            let result = callback(unhandledEvent)
+            
+            if result == .handledSuccessfully {
+                switch unhandledEvent.persistenceRule {
+                case .clearImmediately:
+                    fallthrough
+                case .clearAfterAllCurrentSubscribersNotified:
+                    unhandledEvents.remove(at: index)
+                    
+                case .neverClear:
+                    continue
+                }
+            }
+        }
+        
+        log("Registered listener for identifier \"\(stringFor(id: id) ?? String(describing: id)))\"")
+    }
+    
+    public static func post<T: PersistentEvent>(event: T) {
+        cleanup()
+        
+        let eventType = String(reflecting: T.self)
+        var handled = false
+        
+        iteration: for subscription in subscriptions.filter({ $0.eventType == eventType }) {
```

Comment:
Confused here too

Reply:
The `iteration: ` portion of this loop is simply an identifier for the iteration.  Look below, when we go into a `switch...case` statement.  I need to `break` from the iteration, NOT the switch case.  So I give it an identifier to let the compiler know exactly what I intended to `break`.